tools/toolchains/arm.py

```diff
@@ -175,32 +177,35 @@ def compile_c(self, source, object, includes):
     def compile_cpp(self, source, object, includes):
         return self.compile(self.cppc, source, object, includes)
 
-    @hook_tool
-    def link(self, output, objects, libraries, lib_dirs, mem_map):
-        map_file = splitext(output)[0] + ".map"
-        if len(lib_dirs):
-            args = ["-o", output, "--userlibpath", ",".join(lib_dirs), "--info=totals", "--map", "--list=%s" % map_file]
-        else:
-            args = ["-o", output, "--info=totals", "--map", "--list=%s" % map_file]
-
-        args.extend(self.flags['ld'])
-
-        if mem_map:
-            args.extend(["--scatter", mem_map])
+    def make_real_scatter(self, scatter_file):
```

Comment:
What about leaving the shebang in as is for ARMC5 so all the linker script stay as they are and have make_real_scatter replace the ARMC5 shebang with the ARMC6 shebang only when building for ARMC6?